main.tf

```diff
@@ -108,39 +170,84 @@ resource "google_container_cluster" "cluster" {
 
-  cluster_ipv4_cidr         = null
-  default_max_pods_per_node = var.default_max_pods_per_node
+  dynamic "ip_allocation_policy" {
+    for_each = var.ip_allocation_policy != null ? [var.ip_allocation_policy] : []
 
-  ip_allocation_policy {
-    cluster_secondary_range_name  = var.cluster_secondary_range_name
-    services_secondary_range_name = var.services_secondary_range_name
+    content {
+      cluster_ipv4_cidr_block       = try(ip_allocation_policy.value.cluster_ipv4_cidr_block, null)
+      services_ipv4_cidr_block      = try(ip_allocation_policy.value.services_ipv4_cidr_block, null)
+      cluster_secondary_range_name  = try(ip_allocation_policy.value.cluster_secondary_range_name, null)
+      services_secondary_range_name = try(ip_allocation_policy.value.services_secondary_range_name, null)
+    }
   }
 
-  # TODO: support more options
-  maintenance_policy {
-    daily_maintenance_window {
-      start_time = var.maintenance_start_time
+  dynamic "maintenance_policy" {
+    for_each = var.maintenance_policy != null ? [var.maintenance_policy] : []
+
+    content {
+      dynamic "daily_maintenance_window" {
+        for_each = try([maintenance_policy.value.daily_maintenance_window], [])
+
+        content {
+          start_time = daily_maintenance_window.value.start_time
+        }
+      }
+
+      dynamic "recurring_window" {
+        for_each = try([maintenance_policy.value.recurring_window], [])
+
+        content {
+          start_time = recurring_window.value.start_time
+          end_time   = recurring_window.value.end_time
+          recurrence = recurring_window.value.recurrence
+        }
+      }
+
+      dynamic "maintenance_exclusion" {
+        for_each = try(maintenance_policy.value.maintenance_exclusion, [])
+
+        content {
+          exclusion_name = maintenance_exclusion.value.exclusion_name
+          start_time     = maintenance_exclusion.value.start_time
+          end_time       = maintenance_exclusion.value.end_time
+        }
+      }
     }
   }
 
-  ### Remove default node-pool after initial creation
+  # --------------------------------------------------------------------------------------------------------------------
+  # REMOVE DEFAULT NODE-POOL AFTER INITIAL CREATION
+  # 
   # We create the smallest possible default node-pool and delete it right away
   # there is no way not to create the default node pool.
   # node pools should be created using the terraform-google-gke-node-pool module
-  # for regional clusters thsi will create 1 node in each zone of the region
+  # for regional clusters that will create 1 node in each zone of the region
+  # --------------------------------------------------------------------------------------------------------------------
 
-  # node_pool {}
   initial_node_count       = 1
-  remove_default_node_pool = true
+  remove_default_node_pool = !var.enable_autopilot ? true : null
```

Comment:
where is this relation to autopilot described in detail?

Reply:
resolved

Reply:
removed since we will move autopilot to a different module at some point